### PR TITLE
Introduce QueryObjectBuilder

### PR DESCRIPTION
**Pull request by Ladsgroup**
Simple builder that takes a QueryRepresentation and produce the object
sparqljs would understand.

Currently, the types are not in master yet and as the result, lints fail
for using a general "any" type.

### CODE REVIEW

File: src/sparql/QueryObjectBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		this.queryObject = {
+			queryType: 'SELECT',
+			variables: [],
+			where: [],
+			type: 'query',
+			prefixes: allNamespaces
+		};
+	}
+
+	public buildFromQueryRepresentation( queryRepresentation: any ) {
```
Comment (guergana) [inline src/sparql/QueryObjectBuilder.ts]: maybe this can be changed already? was checking the types and not sure which it corresponds to

**Review by guergana — CHANGES_REQUESTED** [review]
update type pleasee

File: src/sparql/QueryObjectBuilder.ts (hunk 1 above)
Comment (Ladsgroup) [reply src/sparql/QueryObjectBuilder.ts]: Done.

File: tests/unit/sparql/QueryObjectBuilder.spec.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import allNamespaces from '@/sparql/rdfNamespaces';
+import QueryObjectBuilder from '@/sparql/QueryObjectBuilder';
+
+describe( 'QueryObjectBuilder', () => {
+	it( 'einfach', () => {
```
Comment (jakobw) [inline tests/unit/sparql/QueryObjectBuilder.spec.ts]: <_<

**Review by jakobw — CHANGES_REQUESTED** [review]
Looks good. The fact that QueryObjectBuilder isn't working with an immutable query object scares me a little. This could go wrong if we don't keep that in mind, but it's fine for now.

I'm also not 100% clear yet how we want to slice and dice the SPARQL generating code. We for sure want some sort of builder, but I was wondering whether we want the builder to be tied to our QueryRepresentation type, or something more generic. Hard to tell.

Needs a better test name, otherwise good to go from my side! :rocket:

**Comment by Ladsgroup** [comment]
> Looks good. The fact that QueryObjectBuilder isn't working with an immutable query object scares me a little. This could go wrong if we don't keep that in mind, but it's fine for now.

I understand that mutable objects are scary but the current builder pattern is pretty common and is not that bad (as you said, if we keep it in mind). We can revisit and clean it up later if it turns out to be bad.
> 
> I'm also not 100% clear yet how we want to slice and dice the SPARQL generating code. We for sure want some sort of builder, but I was wondering whether we want the builder to be tied to our QueryRepresentation type, or something more generic. Hard to tell.

Same but also, AFAIK it's supposed to be "Simple" query builder so I hope nothing to complicated shows up at our doorstep.
> 
> Needs a better test name, otherwise good to go from my side! rocket

Damn, I was hoping this would goes unnoticed :D

**Review by jakobw — APPROVED** [review]
:cool: :cool: :cool:

**Pull request merged**